### PR TITLE
[runtime] Use MonoError for mono_thread_create_internal

### PR DESCRIPTION
**Pull request by ludovic-henry**

### CODE REVIEW

File: mono/metadata/threadpool-ms.c [hunk 1]
```diff
@@ -728,7 +729,9 @@ worker_try_create (void)
 		counter._.active ++;
 	});
 
-	if ((thread = mono_thread_create_internal (mono_get_root_domain (), worker_thread, NULL, TRUE, 0)) != NULL) {
+	thread = mono_thread_create_internal (mono_get_root_domain (), worker_thread, NULL, TRUE, 0, &error);
+
+	if (thread && mono_error_ok (&error)) {
```
Comment (kumpera) [inline mono/metadata/threadpool-ms.c]: This is a code smell, there must be only one way to check for an ok result.

Besides that, there must be a compiling reason for a function to return NULL and not be an error.

By not sorting out those issues, this commit makes things worse.

File: mono/metadata/threads.c [hunk 2]
```diff
@@ -568,33 +568,37 @@ set_current_thread_for_domain (MonoDomain *domain, MonoInternalThread *thread, M
...
+	mono_error_raise_exception (&error); /* FIXME don't raise here */
+
 	MONO_OBJECT_SETREF (thread, internal_thread, internal);
+
 	return thread;
 }
 
 static MonoInternalThread*
-create_internal_thread (void)
+create_internal_thread (MonoError *error)
```
Comment (kumpera) [inline mono/metadata/threads.c]: This function is changing two functions instead of one.

The commit message must be fixed, at least.

Comment (ludovic-henry) [inline mono/metadata/threads.c]: It is already split in 2 commits, `create_internal_thread` is modified in https://github.com/ludovic-henry/mono/commit/71c1a8ad9befbc7391018c74e4e771fb3dfb1ee9

**Comment by kumpera** [comment]
We still need to make sure mono_thread_create_internal return value is NULL iff an error is set and the reciprocal for non null returns.

**Comment by ludovic-henry** [comment]
So out of date, it's easier to close it and restart.

**Comment by ludovic-henry** [comment]
So out of date, it's easier to close it and restart.

**Pull request closed without merging**